server.go

```diff
@@ -296,7 +303,12 @@ func WithPort(port int) func(*Server) {
 // WithAddr optionally specifies the TCP address for the server to listen on, in the form "host:port".
 // If not specified addr ':9999' will be used.
 func WithAddr(addr string) func(*Server) {
-	return func(c *Server) { c.Server.Addr = addr }
+	return func(c *Server) {
+		if c.listener != nil {
+			panic("cannot set addr when a listener is already configured")
```

Comment:
I think we're better off documenting the behavior rather than handling it. Most users will never need to reach the level of passing their own listener. When you reach that point you tend to really know what you're getting yourself into. Would add a note stating that if `WithListener` is used that `WithAddr` is not respected.